app/js/components/tags-page.js

```diff
@@ -7,10 +7,42 @@
 'use strict';
 
 import React from 'react';
-
 import TagsList from './tags-list';
 
 export default class TagsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {tags: []};
+  }
+
+  componentDidMount() {
+    this.getTagList();
+  }
+
+  componentDidUpdate() {
+    this.getTagList();
+  }
+
+  getTagList() {
+    // TODO: make this an api-client call
+    const tagsRequest = new Promise((resolve, reject) => {
+      const xhr = new XMLHttpRequest();
+      xhr.open('GET', '/qdb/sample/tags.json');
+
+      xhr.onload = () => {
+        if (xhr.status == 200) {
+          resolve(JSON.parse(xhr.responseText));
+        } else {
+          reject('an error has occured');
+        }
+      };
+
+      xhr.send();
+    });
+
+    tagsRequest.then(tags => this.setState({...this.state, tags}));
+  }
+
```

Comment:
You shouldn't really be testing your application using the production API anyway.